cloud-controller-manager/do/lb_service_admission_handler.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2023 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package do
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/digitalocean/godo"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// LBServiceAdmissionHandler validates service type LB.
+type LBServiceAdmissionHandler struct {
+	log        *logr.Logger
+	godoClient *godo.Client
+
+	decoder   *admission.Decoder
+	region    string
+	clusterID string
+	vpcID     string
+}
+
+// NewLBServiceAdmissionHandler returns a configured instance of LBServiceHandler.
+func NewLBServiceAdmissionHandler(log *logr.Logger, godoClient *godo.Client) *LBServiceAdmissionHandler {
+	return &LBServiceAdmissionHandler{
+		log:        log,
+		godoClient: godoClient,
+		decoder:    admission.NewDecoder(runtime.NewScheme()),
+	}
+}
+
+// Handle handles admissions requests for load balancer services.
+func (h *LBServiceAdmissionHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	resp := h.handle(ctx, req)
+
+	logFields := []any{"object_name", req.Name, "object_namespace", req.Namespace, "object_kind", req.Kind.String()}
+	if resp.Allowed {
+		h.log.Info("allowing admission request", logFields...)
```

Comment:
I wonder if we should use `Debug` here given busy clusters could be receiving a large number of requests should nodes come and up frequently (e.g., with autoscaling clusters), which in turn could "spam" the logs. 🤔 No super strong opinions though. Any thoughts?

Reply:
I initially added that log message because I found it interesting to see the passed vs failed requests, but I agree it can become very noisy. I'm going to set it to `Debug` and work on a subsequent PR for metrics reporting. Reporting `Allowed`/`Denied`/`Errored` admissions per endpoint should do the trick.

Reply:
It's worth double-checking what kind of metrics may already be produced by controller-runtime. Best case, we don't have to do anything in this regard, but I'm not certain. We'll have to check.

Reply:
I looked at what metrics were collected yesterday, and I was a bit surprised that webhook-specific metrics didn't seem to report status: [pkg/webhook/internal/metrics/metrics.go](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/webhook/internal/metrics/metrics.go). 

It will require a bit of work, considering that metrics aren't served by default. I have a beginning of something stashed.